### ISSUE TITLE
Support for HTTP requests/s

### ISSUE THREAD

**Issue by pgmsft** [Microsoft/ethr#57]
This needs work to do HTTP requests from multiple threads to see how many requests/s it can push. This would be useful for testing performance of middle boxes like firewalls, proxies etc.

**Comment by boyan-soubachov** [comment]
I'd be happy to take this on as a first issue. I'm guessing the problem addressed by this issue is to effectively incorporate network load testing of middle boxes through the use of multi-threading?

**Comment by pgmsft** [comment]
Yes, this is for middle box testing. Sure, please take over and submit a PR.

**Comment by boyan-soubachov** [comment]
Excellent! I just want to get your thoughts in terms of solution design:

I see the client already supports multi-threaded connections for the HTTP and HTTPS bandwidth tests. What did you exactly have in mind with this issue?

**Comment by pgmsft** [comment]
Client is doing bandwidth based testing. For requests/s testing, it should generate requests from multiple threads of whatever size is specified by the user, and measure how many requests/s it is doing. The change should be relatively simpler and primarily around doing the correct measurements related to requests/s. For example, see difference between TCP bandwidth based tests and CPS based tests.

**Comment by boyan-soubachov** [comment]
I see, that clears things up a lot; thank you! :+1:

**Comment by boyan-soubachov** [comment]
P.S. Is there a reason you've kept everything in the `main` package or would you be happy for me to also start refactoring things? 

I'm thinking of initially moving the client/s into their own package which can be imported by `main`.

**Comment by pgmsft** [comment]
I am ok with refactoring. This was my first golang project, so that is the only reason for everything in one package :)
